File: QytetetRuby/pkg/QytetetRuby-0.0.1/lib/casilla.rb
#encoding: utf-8
# Francisco Beltrán Sánchez
# Pablo Cordero Romero
module ModeloQytetet
  class Casilla
    attr_reader :numeroCasilla
    attr_accessor :precioCompra

    def initialize(numCas, coste)
      @numeroCasilla = numCas
      @precioCompra = coste
    end
    
    def self.copia(casilla)
      self.new(casilla.numeroCasilla, casilla.precioCompra)
    end

    # Dejamos este método ya que no podemos modificar precioCompra desde calle
    def setTitulo(ttl)
      @precioCompra = ttl.precioCompra
    end

    def soyEdificable; end
    def tengoPropietario; end

    def to_s
        "\n-----\nCasilla: \n\tnumeroCasilla= #{@numeroCasilla} \n\tprecioCompra= #{@precioCompra}";
    end
    
    private :setTitulo
  end
end

File: QytetetJava/src/modeloqytetet/Casilla.java
/**
 * Practica de:
 * 
 * Francisco Beltrán Sánchez
 * Pablo Cordero Romero
 */
package modeloqytetet;

public abstract class Casilla {
    private int numeroCasilla = 0;
    private int coste;

    // Constructores con parámetros para casilla
    Casilla(int numeroCasilla, int coste){
        this.coste = coste;
        this.numeroCasilla = numeroCasilla;
    }

    // Constructor copia casilla
//    Casilla(Casilla casilla){
//        this.numeroCasilla = casilla.numeroCasilla;
//        this.coste = casilla.coste;
//    }   

    int getNumeroCasilla() {
        return numeroCasilla;
    }

    int getCoste() {
        return coste;
    }

    public void setCoste(int coste){
        this.coste = coste;
    }

    protected abstract TituloPropiedad getTitulo();
    protected abstract TipoCasilla getTipo();
    protected abstract boolean soyEdificable();
    public abstract boolean tengoPropietario();

    @Override
    public String toString() {
        return "\n-----\nCasilla:" + "\n\tnumeroCasilla=" + numeroCasilla + "\n\tprecioCompra=" + coste;
    }
    
}
